components/BreadCrumbs.vue

```diff
@@ -0,0 +1,87 @@
+<script>
+import { pluralize } from '../utils/string';
+export default {
+  data() {
+    // make a map of all route names to validate programatically generated names
+    const allRoutes = this.$router.options.routes;
+    const allRouteMap = allRoutes.reduce((all, route) => {
+      all[route.name] = route;
+
+      return all;
+    }, {});
+    const { name, params } = this.$route;
+
+    const crumbPieces = name.split('-');
+
+    let crumbLocations = [];
+
+    crumbPieces.forEach((piece, i) => {
+      let nextName = piece;
+
+      if (crumbLocations[i - 1]) {
+        nextName = ( `${ crumbLocations[i - 1].name }-${ piece }`);
+      }
+      crumbLocations.push({
+        name:   nextName,
+        params: this.paramsFor(nextName, params)
+      });
+    });
+
+    // remove root route 'c'
+    crumbLocations.shift();
+
+    // filter invalid routes
+    crumbLocations = crumbLocations.filter((location) => {
+      return (allRouteMap[location.name] && this.displayName(location, params));
+    });
+
+    return {
+      crumbLocations, params, crumbPieces, allRouteMap
+    };
+  },
+
+  methods: {
+    paramsFor(crumbName, params = this.params) {
+      const pieces = crumbName.split('-');
+      const out = {};
+
+      pieces.forEach((piece) => {
+        if (params[piece]) {
+          out[piece] = params[piece];
+        }
+      });
+
+      return out;
+    },
+
+    displayName(location, params = this.params) {
+      const pieces = location.name.split('-');
+      const lastPiece = pieces[pieces.length - 1];
+
+      if (lastPiece === 'resource') {
+        return pluralize(null, params[lastPiece]);
```

Comment:
I think this is going to produce the wrong result for types that are virtual or remapped by type-config/type-map?  e.g. for Endpoints the actual type is `endpoints` instead of `endpoint`, so it's remapped, but the param is the actual original value, and you'll end up with `endpointses`.

Ideally for the pieces that are a resource you'd use the type-map store's `pluralLabelFor` getter.